src/validation.ts

```diff
@@ -0,0 +1,16 @@
+import {SignedState} from '.';
+import {ethers} from 'ethers';
+import * as forceMoveApp from './contract/force-move-app';
+
+export async function validTransition(
+  signer: ethers.Signer,
+  newState: SignedState,
+  previousStates: SignedState[],
```

Comment:
Why does this accept an array of previous states, when only the last one is ever used? It could also work like this:

```
  previousState?: SignedState
```

Reply:
The idea is that the caller can blindly pass in their history of `signedState`s and not have to bother picking out the last one. But perhaps explicitly taking in a `previousState` might be clearer.